File: atmos_mixer_pro/fix_engine.dart
import 'dart:io';

void main() {
  final file = File('lib/features/exhibition/state/three_js_engine_provider.dart');
  String content = file.readAsStringSync();

  content = content.replaceFirst(
    '''  void executeJavaScript(String js) {
    if (_isEngineReady && _webViewController != null) {''',
    '''  void executeJavaScript(String js) {
    if (isEngineReady && _webViewController != null) {'''
  );

  content = content.replaceFirst(
    '''  @override
  void dispose() {''',
    '''  void dispose() {'''
  );

  file.writeAsStringSync(content);
}

File: atmos_mixer_pro/fix_engine.py
import re

def main():
    path = "lib/features/exhibition/state/three_js_engine_provider.dart"
    with open(path, "r") as f:
        content = f.read()

    new_method = """
  void setEarLevel(double level) {
    executeJavaScript("window.updateEarLevel($level);");
  }

  void dispose() {"""
  
    content = content.replace("  void dispose() {", new_method)

    with open(path, "w") as f:
        f.write(content)

main()
